lib/Library.php

```diff
@@ -649,6 +649,24 @@ public static function GetEbook(?string $ebookWwwFilesystemPath): ?Ebook{
 		}
 	}
 
+	public static function GetEbookByIdentifier(?string $identifier): ?Ebook{
```

Comment:
Looking at this again, this function should be refactored into `Ebook::GetByIdentifier(?string $identifier): Ebook` which throws an `EbookNotFoundException` instead of returning `null`, for consistency in the codebase. We should aim to eventually obsolete the `Library` class as it doesn't match the patterns we're using in the rest of the codebase. (It was designed at the start of the project, ~10 years ago, so things have changed significantly since then.)

You don't have to do this right now if you don't have time, it's probably a bigger effort than a quick rename, but eventually, that's how it should look eventually.

Reply:
It was no prob, and a good call. Done.